chiselFrontend/src/main/scala/chisel3/Bits.scala

```diff
@@ -408,30 +380,67 @@ sealed class UInt private[chisel3] (width: Width) extends Bits(width) with Num[U
   private[chisel3] override def cloneTypeWidth(w: Width): this.type =
     new UInt(w).asInstanceOf[this.type]
 
-  // TODO: refactor to share documentation with Num or add independent scaladoc
-  /** Unary negation (expanding width)
-    *
-    * @return a $coll equal to zero minus this $coll
-    * $constantWidth
-    * @group Arithmetic
-    */
-  final def unary_- (): UInt = macro SourceInfoTransform.noArg
-
-  /** Unary negation (constant width)
-    *
-    * @return a $coll equal to zero minus this $coll shifted right by one.
-    * $constantWidth
-    * @group Arithmetic
-    */
-  final def unary_-% (): UInt = macro SourceInfoTransform.noArg
-
   /** @group SourceInfoTransformMacro */
   def do_unary_- (implicit sourceInfo: SourceInfo, compileOptions: CompileOptions) : UInt = 0.U - this
   /** @group SourceInfoTransformMacro */
   def do_unary_-% (implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): UInt = 0.U -% this
 
-  override def do_+ (that: UInt)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): UInt = this +% that
-  override def do_- (that: UInt)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): UInt = this -% that
+  /** Static left shift operator
+   *
+   * @param that an amount to shift by
+   * @return this $coll with `that` many zeros concatenated to its least significant end
+   * $sumWidthInt
+   * @group Bitwise
+   */
+  override def << (that: BigInt): UInt = macro SourceInfoWhiteboxTransform.thatArg
```

Comment:
Can this be converted to a `SourceInfoTransform` blackbox macro now since we don't need to refine the return type? 

Whereas before this was defined in `Bits`, this is now implementing a typeclass.

Same comment for all the other methods implementing `NumBits`.

Reply:
Turns out we can eliminate *all* uses of `SourceInfoWhiteboxTransform` in the codebase. Should we remove it all-together or comment it out?

Reply:
Orthogonally: I would prefer to never comment stuff out (just remove it). I think that `SourceInfoWhiteboxTransform` can then be removed, but I would like to check with @ducky64 and @jackkoenig.